packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -0,0 +1,195 @@
+      'Wild' +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  } else if (split.at(-1)?.startsWith('[')) {
+    const fileName = split
+      .at(-1)!
+      .replace('.tsx', '')
+      .replace('[', '')
+      .replace(']', '');
+    return (
+      split.slice(0, -1).join('') +
+      'Slug' +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  } else {
+    const fileName = split.at(-1)!.replace('.tsx', '');
+    return (
+      split.slice(0, -1).join('') +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  }
+};
+
+export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
+  let entriesFile: string | undefined;
+  let pagesDir: string | undefined;
+  let outputFile: string | undefined;
+  let formatter = (s: string): Promise<string> => Promise.resolve(s);
+  return {
+    name: 'vite-plugin-fs-router-typegen',
+    apply: 'serve',
+    async configResolved(config) {
+      pagesDir = joinPath(config.root, opts.srcDir, SRC_PAGES);
+      entriesFile = joinPath(config.root, opts.srcDir, `${SRC_ENTRIES}.tsx`);
+      outputFile = joinPath(config.root, opts.srcDir, `${SRC_ENTRIES}.gen.tsx`);
+
+      try {
+        const prettier = await import('prettier');
+        // Get user's prettier config
+        const config = await prettier.resolveConfig(outputFile);
+
+        formatter = (s) =>
+          prettier.format(s, { ...config, parser: 'typescript' });
+      } catch {
+        // ignore
+      }
+    },
+    configureServer(server) {
+      if (!entriesFile || !pagesDir || !outputFile || existsSync(entriesFile)) {
+        return;
+      }
+
+      // Recursively collect `.tsx` files in the given directory
+      const collectFiles = async (dir: string): Promise<string[]> => {
+        if (!pagesDir) return [];
+        let results: string[] = [];
+        const files = await readdir(dir, { withFileTypes: true });
+
+        for (const file of files) {
+          const fullPath = path.join(dir, file.name);
+          if (file.isDirectory()) {
+            results = results.concat(await collectFiles(fullPath));
+          } else if (file.isFile() && fullPath.endsWith('.tsx')) {
+            results.push(fullPath.replace(pagesDir, ''));
+          }
+        }
+        return results;
+      };
+
+      const fileExportsGetConfig = (filePath: string) => {
+        if (!pagesDir) return false;
+        const file = readFileSync(pagesDir + filePath).toString();
+
+        return (
+          file.includes('const getConfig') ||
+          file.includes('function getConfig')
+        );
+      };
+
+      const generateFile = (filePaths: string[]): string => {
+        const fileInfo = [];
+        for (const filePath of filePaths) {
+          // where to import the component from
+          const src = filePath.slice(1);
+          const hasGetConfig = fileExportsGetConfig(filePath);
+
+          if (filePath.endsWith('_layout.tsx')) {
```

Comment:
not sure if it has the same issue with `/pages/foo_layout.tsx`.

Reply:
yep! fixed, thanks